### DISCUSSION TITLE
REST device service - Storage of image file in core-data

### DISCUSSION THREAD

**Discussion by sudhamani-hcl** [edgexfoundry/community#75] [General]
Hi Team,

We are using REST device service for the reception of the image files. We used auto-events to receive the image file for every 30s from the end device. 

After the reception of image file from the end device, we tried to read the image file from the core-data using the command `curl -X 'GET' 'http://localhost:59880/api/v2/reading/device/name/2way-rest-device/resourceName/jpeg'`, we are not able to see the image file(binary data) in the output. The "value" field in the output was empty as shown below:
`{"apiVersion":"v2","statusCode":200,"totalCount":1,"readings":[{"id":"9fcf9f78-3ef9-4f81-b520-f89702b65ffd","origin":1679398138447587695,"deviceName":"2way-rest-device","resourceName":"jpeg","profileName":"sample-2way-rest-device","valueType":"Binary","mediaType":"image/jpeg","value":""}]}`

Please let us know the received image data is stored in the core-data or not. If it is not getting stored in the core-data, then where else it will be stored?

Thanks in advance,
Sudhamani

**Comment by cloudxxx8** [comment]
Due to the nature of the core-data persistence layer, we don't store binary data in core-data.
Currently, the persistence service is Redis, and it's a memory database.  Storing binary data into it would cause out of memory easily.
The binary reading can be subscribed in the message bus anyway, so the solution is to write an app service to handle binary reading and store to a proper place you want, for example, PostgreSQL is fine to handle it.

**Comment by sudhamani-hcl** [comment]
Thank you @cloudxxx8
